iap/src/main/java/com/example/iap/VerifyIapRequestHeader.java

```diff
@@ -96,38 +57,20 @@ boolean verifyJwtForComputeEngine(
             Long.toUnsignedString(projectNumber), Long.toUnsignedString(backendServiceId)));
   }
 
-  private boolean verifyJwt(String jwtToken, String expectedAudience) throws Exception {
-
-    // parse signed token into header / claims
-    SignedJWT signedJwt = SignedJWT.parse(jwtToken);
-    JWSHeader jwsHeader = signedJwt.getHeader();
-
-    // header must have algorithm("alg") and "kid"
-    Preconditions.checkNotNull(jwsHeader.getAlgorithm());
-    Preconditions.checkNotNull(jwsHeader.getKeyID());
-
-    JWTClaimsSet claims = signedJwt.getJWTClaimsSet();
-
-    // claims must have audience, issuer
-    Preconditions.checkArgument(claims.getAudience().contains(expectedAudience));
-    Preconditions.checkArgument(claims.getIssuer().equals(IAP_ISSUER_URL));
-
-    // claim must have issued at time in the past
-    Date currentTime = Date.from(Instant.now(clock));
-    Preconditions.checkArgument(claims.getIssueTime().before(currentTime));
-    // claim must have expiration time in the future
-    Preconditions.checkArgument(claims.getExpirationTime().after(currentTime));
-
-    // must have subject, email
-    Preconditions.checkNotNull(claims.getSubject());
-    Preconditions.checkNotNull(claims.getClaim("email"));
-
-    // verify using public key : lookup with key id, algorithm name provided
-    ECPublicKey publicKey = getKey(jwsHeader.getKeyID(), jwsHeader.getAlgorithm().getName());
-
-    Preconditions.checkNotNull(publicKey);
-    JWSVerifier jwsVerifier = new ECDSAVerifier(publicKey);
-    return signedJwt.verify(jwsVerifier);
+  private boolean verifyJwt(String jwtToken, String expectedAudience) {
+    TokenVerifier tokenVerifier = TokenVerifier.newBuilder()
+        .setAudience(expectedAudience)
+        .setIssuer(IAP_ISSUER_URL)
+        .build();
+    try {
+      JsonWebToken jsonWebToken = tokenVerifier.verify(jwtToken);
+
+      // Verify that the token contain subject and email claims
+      JsonWebToken.Payload payload = jsonWebToken.getPayload();
+      return payload.getSubject() != null && payload.get("email") != null;
+    } catch (TokenVerifier.VerificationException e) {
```

Comment:
Do we want to print the exception message so the user knows why verification failed?

Reply:
Sure makes sense - updated to print the message to stdout.